src/pages/apps.tsx

```diff
@@ -37,6 +37,7 @@ const Apps: NextPage = () => {
     permissions: safeApp?.safeAppsPermissions || [],
     addPermissions,
     getPermissions,
+    remoteSafeAppsLoading: remoteSafeAppsLoading || remoteSafeApps.length === 0,
```

Comment:
What if remoteSafeApps.length is legit 0?

Reply:
I agree it's hacky, but I don't see any other way to detect a state where the apps aren't loaded yet and the request hadn't been triggered. We don't have an environment with 0 safe apps, at least "native" ones are available on any env.

Reply:
The ultimate solution is to refactor from using boolean variables to determine request state to a state machine with clear status ("not loaded", "loading", "loaded", error"), but since it uses an app-wide hook that's too much of a refactor at this time.

Reply:
Shouldn’t remoteSafeApps be completely undefined prior to the request?

Reply:
I'd would help in this particular case but on a global scale it's also a hacky solution

Reply:
you'd still have to write clumsy stuff like `typeof remoteSafeApps !== 'undefined' && !loading && !error` to determine request's state + have a complexity of checking for undefined everywhere before accessing

Reply:
You don’t need to use typeof, the variable is always defined. Hacky or not, it’s less wrong than checking for length.

Reply:
What's wrong with checking for length if you factor in that we don't have a single environment with 0 apps and thus an array can't be empty if the list is loaded?

Reply:
Actually, a fair argument for your solution would be interface works. We can't guarantee that forked environments will contain apps. I'm going on vacation for 2 weeks, I'll give a heads-up to the team to iterate on this PR.

Reply:
I've made a commit to this branch. Please feel free to revert or adjust it (whoever overtakes this PR).